orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh

```diff
@@ -71,25 +71,47 @@ PATH_PREFIX="/user"
 
 export KRB5CCNAME=/tmp/krb5cc_cloudbreak_$EUID
 
+
+klist -s
+if [[ $? -ne 0 ]]; then
+  kinit_as_hdfs
+fi
+
 WEBHDFS_HTTP_POLICY=$(hdfs getconf -confkey dfs.http.policy)
 
-if [ "$WEBHDFS_HTTP_POLICY" == "HTTP_ONLY" ]; then
-  NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.http-address)
-  WEBHDFS_URL=http://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+hdfs haadmin -getAllServiceState
+if [[ $? -ne 0 ]]; then
+  echo "Single node configuration"
+  if [ "$WEBHDFS_HTTP_POLICY" == "HTTP_ONLY" ]; then
+    NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.http-address)
+    WEBHDFS_URL=http://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  else
+    NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.https-address)
+    WEBHDFS_URL=https://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  fi
 else
-  NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.https-address)
-  WEBHDFS_URL=https://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  echo "HA configuration"
+  NN_HOST=$(hdfs haadmin -getAllServiceState | grep active | cut -d':' -f 1)
+  echo "Active NAMENODE host: $NN_HOST"
+  if [ "$NN_HOST" != "$(hostname -f)" ]; then
```

Comment:
Do we need this check in the case of non HA? This to me looks like this script will be installed on all the nodes or maybe just the namenodes?

Reply:
this will be on every, but only namenodes, and will run there, so if they switch the active one for any reason we don't have to do anything from cb side